app/pages/lab/social-links-editor.jsx

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import DragReorderable from 'drag-reorderable';
+import AutoSave from '../../components/auto-save.coffee';
+import SOCIAL_ICONS from '../../lib/social-icons';
+
+export default class SocialLinksEditor extends React.Component {
+  constructor(props) {
+    super(props);
+    const socialOrder = Object.keys(SOCIAL_ICONS).map(key => key);
+    this.state = {
+      socialOrder
+    };
+    this.reorderDefault = this.reorderDefault.bind(this);
+    this.handleNewLink = this.handleNewLink.bind(this);
+    this.handleLinkReorder = this.handleLinkReorder.bind(this);
+    this.handleRemoveLink = this.handleRemoveLink.bind(this);
+    this.renderRow = this.renderRow.bind(this);
+  }
+
+  componentDidMount() {
+    this.reorderDefault();
+  }
+
+  reorderDefault() {
+    const socialUrls = this.props.project.urls.filter(url => url.path);
+    const newOrder = [];
+    socialUrls.map(link => newOrder.push(link.site));
+    this.state.socialOrder.map((item) => {
+      if (newOrder.indexOf(item) < 0) {
+        newOrder.push(item);
+      }
+    });
+    this.setState({ socialOrder: newOrder });
+  }
+
+  handleNewLink(site, e) {
+    let index = this.indexFinder(this.props.project.urls, site);
+    if (index < 0) { index = this.props.project.urls.length; }
+
+    if (e.target.value) {
+      const changes = {
+        [`urls.${index}`]: {
+          label: '',
+          path: e.target.value,
+          site,
+          url: `https://${site}${e.target.value}`
+        }
+      };
+      this.props.project.update(changes);
+    } else {
+      this.handleRemoveLink(site);
+    }
+  }
+
+  handleLinkReorder(newLinkOrder) {
+    const socialUrls = [];
+    const urls = this.props.project.urls.slice();
+    let index = urls.length - 1;
+    while (index >= 0) {
+      if (urls[index].path) {
+        socialUrls.push(urls.splice(index, 1)[0]);
+      }
+      index -= 1;
+    }
+    newLinkOrder.map((item) => {
+      const urlIndex = this.indexFinder(socialUrls, item);
+      if (urlIndex >= 0) {
+        urls.push(socialUrls[urlIndex]);
+      }
+    });
+    const changes = { urls };
+    this.props.project.update(changes).save();
+    this.setState({ socialOrder: newLinkOrder });
+  }
+
+  handleRemoveLink(linkToRemove) {
+    const urls = this.props.project.urls.slice();
+    const indexToRemove = this.indexFinder(urls, linkToRemove);
+    if (indexToRemove > -1) {
+      urls.splice(indexToRemove, 1);
+      this.props.project.update({ urls }).save();
+    }
+  }
+
+  handleDisableDrag(event) {
+    event.target.parentElement.parentElement.setAttribute('draggable', false);
+  }
+
+  handleEnableDrag(event) {
+    event.target.parentElement.parentElement.setAttribute('draggable', true);
+  }
+
+  indexFinder(toSearch, toFind) { //eslint-disable-line
```

Comment:
You're right about why I included this disabled line, but it's not consistent that I have this disabled line in only one of the methods. Probably a good idea to remove the comment. This is probably worth bringing up in a channel (how to handle these static methods, or removing the linter rule).